cmd/main.go

```diff
@@ -35,9 +43,42 @@ var (
 	}, nil)
 )
 
+
+func initTracerProvider(exporterType string) *sdktrace.TracerProvider {
+	resources, _ := resource.New(context.Background(),
+		resource.WithFromEnv(), // pull attributes from OTEL_RESOURCE_ATTRIBUTES and OTEL_SERVICE_NAME environment variables
+		resource.WithProcess(), // This option configures a set of Detectors that discover process information
+	)
+	otel.SetTextMapPropagator(propagation.NewCompositeTextMapPropagator(propagation.TraceContext{}))
+	if exporterType != "" {
+		exporter := exporterTracing(exporterType)
+		tp := sdktrace.NewTracerProvider(
+			sdktrace.WithResource(resources),
+			sdktrace.WithBatcher(exporter),
+		)
+		otel.SetTracerProvider(tp)
+		return tp
```

Comment:
nit: this does not read well - e.g. to have two return statements in this function. 

It would be easier to assign `trace.NewNoopTracerProvider()` to a variable and override it in this block if the exporter type is not empty.

Reply:
this is not comming from same packages one is sdktrace the other is just trace